src/Specular/Internal/Incremental/PriorityQueue.js

```diff
@@ -0,0 +1,88 @@
+// new :: forall a.
+//   EffectFn4
+//     (Optional Any)                 -- Optional.none
+//     (Field a Mutable Int)          -- Priority
+//     (Field a Mutable Boolean)      -- Is the entry present in this queue?
+//     (Field a Mutable (Optional a)) -- Next entry with the same priority
+//   (PQ a)
+exports.new = function(none, priorityField, presentField, nextField) {
+  return {
+    none: none,
+    priorityField: priorityField,
+    presentField: presentField,
```

Comment:
`present` in the sense of `current` as opposed to `next`? It's like a cursor?

Reply:
Ok, I get it now, these are the names of fields where queue items store stuff needed by the queue. Deserves a line of comment.

Reply:
Lol, the comments are there! Nevermind,

Reply:
Note: these were only added because I initially thought we'd need two queues (like the original Incremental). Now that we know only one is needed, what do you think about specializing `PriorityQueue` to `SomeNode`, and removing this indirection?